src/app/core/actions/ports/PortChangeAction.ts

```diff
@@ -16,7 +16,9 @@ export abstract class PortChangeAction implements Action {
 
     private wireDeletionAction: GroupAction;
 
-    protected constructor(designer: CircuitDesigner, target: number, initialCount: number) {
+    protected constructor(designer: CircuitDesigner | undefined, target: number, initialCount: number) {
+        if (designer === undefined)
```

Comment:
Prefer `if (!designer)` but also why is this needed? why did you make the parameter `| undefined`?

Reply:
Subclasses call `super(obj.getDesigner(), target, initial);` and `obj.getDesigner()` can return undefined